app/api/events.py

```diff
@@ -70,6 +70,13 @@ def validate_event(user, modules, data):
         raise ForbiddenException({'source': ''},
                                  "Only verified accounts can publish events")
 
+    if data.get('state', None) == 'published':
+        event = Event.query.filter_by(name=data.get('name', None)).first()
```

Comment:
@kushthedude this will simply break in `before_post` call when there is no event to query.

Reply:
@uds5501 Where should it be put first ?

Reply:
@kushthedude IMO it shouldn't be placed in before_post at least. Try and test it. See if it even works.